src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -86,8 +86,28 @@ public async Task<(JObject containerObject, ArraySegment<string> remaining)> Res
             if (methodInfo == null)
                 return (null, null);
 
+            string[] parts;
+            if (includeFullName)
+            {
+                string fullName = methodInfo.IsAsync == 0 ? methodInfo.TypeInfo?.FullName : StripAsyncPartOfFullName(methodInfo.TypeInfo?.FullName);
+                string[] fullNameParts = fullName.Split(".", StringSplitOptions.TrimEntries);
+                int overlappingPartIdx = Enumerable.Range(0, fullNameParts.Length).LastOrDefault(i => fullNameParts[i] == expressionParts[0]);
```

Comment:
Also, this is trying to match starting at the first part of the full name, but I think going from inside to out might be correct. Can you figure out a test to check whether I'm correct or not?